src/Windows/Host/Client/Impl/BrokerServices/StaticFileServerBase.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.Common.Core.IO;
+using Microsoft.Common.Core.Logging;
+using static System.FormattableString;
+
+namespace Microsoft.R.Host.Client.BrokerServices {
+    internal abstract class StaticFileServerBase {
+        private static Random _random = new Random();
+        private HttpListener _listener;
+
+        protected IFileSystem FileSystem { get; }
+        protected IActionLog Log { get; }
+        protected IConsole Console { get; }
+        protected HttpListener Listener => _listener;
+
+        public StaticFileServerBase(IFileSystem fs, IActionLog log, IConsole console) {
+            FileSystem = fs;
+            Log = log;
+            Console = console;
+        }
+
+        protected async Task<bool> InitializeAsync(CancellationToken ct) {
+            await TaskUtilities.SwitchToBackgroundThread();
+
+            _listener = _listener ?? new HttpListener();
+            if (_listener != null && _listener.IsListening) {
```

Comment:
`_listener != null` is always going to be true, since a new instance is created on the line above if it was null when this method was entered.